pkg/watch/versioned/decoder.go

```diff
@@ -14,53 +14,59 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package json
+package versioned
 
 import (
 	"encoding/json"
 	"fmt"
 	"io"
 
 	"k8s.io/kubernetes/pkg/runtime"
+//	"k8s.io/kubernetes/pkg/runtime/serializer/streaming"
 	"k8s.io/kubernetes/pkg/watch"
 )
 
 // Decoder implements the watch.Decoder interface for io.ReadClosers that
```

Comment:
I might suggest making this not depend on versioned - instead, you can depend on the internal watch type and do a decode into (the local streaming serializer should handle converting versioned.Event to internal.Event)